sympy/matrices/expressions/matmul.py

```diff
@@ -388,8 +389,23 @@ def combine_one_matrices(mul):
 
     return newmul(factor, *new_args)
 
+def simp_expr(mul):
```

Comment:
This can be too much ambiguous name to only implement binary distributive law with monomials.
I'd rename like `distribute_monom`